apps/web/src/Providers.tsx

```diff
@@ -22,28 +24,41 @@ const queryClient = new QueryClient({
   },
 });
 
+/** Full-page loader that uses color-preferences for background */
+const fallbackDisplay = (
+  <div
+    className={css({
+      h: '100dvh',
+      w: '100dvw',
+      display: 'grid',
+      placeItems: 'center',
+      bg: 'surface.page',
+      // Root element may not have loaded so rely on OS
+      _osDark: { bg: 'legacy.BGDark' },
+      _osLight: { bg: 'legacy.BGLight' },
+    })}
+  >
+    <Loader size={64} variant="bars" color={colors.gradientMiddle} />
+  </div>
+);
+
 /**
  * Centralized Provider hierarchy.
  */
 const Providers: React.FC<PropsWithChildren<{}>> = ({ children }) => {
   return (
     <SegmentProvider>
-      <HelmetProvider>
-        <BrowserRouter basename={CONTEXT_PATH}>
-          <QueryClientProvider client={queryClient}>
-            <AuthProvider>{children}</AuthProvider>
-          </QueryClientProvider>
-        </BrowserRouter>
-      </HelmetProvider>
+      <QueryClientProvider client={queryClient}>
+        <AuthProvider>
+          <LaunchDarklyProvider fallbackDisplay={fallbackDisplay}>
+            <HelmetProvider>
+              <BrowserRouter basename={CONTEXT_PATH}>{children}</BrowserRouter>
+            </HelmetProvider>
+          </LaunchDarklyProvider>
+        </AuthProvider>
+      </QueryClientProvider>
     </SegmentProvider>
   );
 };
 
-export default Sentry.withProfiler(
-  withLDProvider({
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** Moved LD provider to its own file in `shared-web` with logic‏

Reply:
Most of our other similar providers are there, and many of the related behaviors / imports (i.e. `useFeatureFlags`) are there, so it seemed best IMO

Reply:
Sounds good, thanks for the feedback! Updated